src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/ConsoleDebugTargetsProvider.cs

```diff
@@ -0,0 +1,322 @@
+                {
+                    executable = Path.GetFullPath(Path.Combine(workingDir, executable));
+                }
+            }
+
+            // Now validate the executable path and working directory exist
+            ValidateSettings(executable, workingDir, resolvedProfile.Name);
+            
+            // Now get final exe and args. CTtrl-F5 wraps exe in cmd prompt
+            string finalExecutable, finalArguments;
+            GetExeAndArguments(useCmdShell, executable, arguments, out finalExecutable, out finalArguments);
+
+
+            // Do env variables if any
+            // We always want to set our defaults first and then set the profile ones. This permits a 
+            // profile to override ours
+
+            if (resolvedProfile.EnvironmentVariables != null && resolvedProfile.EnvironmentVariables.Count > 0)
+            {
+                foreach(var kvp in resolvedProfile.EnvironmentVariables)
+                {
+                    settings.Environment[kvp.Key] = kvp.Value;
+                }
+            }
+
+            settings.Executable = finalExecutable;
+            settings.Arguments = finalArguments;
+            settings.CurrentDirectory = workingDir;
+            settings.LaunchOperation = DebugLaunchOperation.CreateProcess;
+            settings.LaunchDebugEngineGuid = await GetDebuggingEngineAsync().ConfigureAwait(false); //.GetDebugEngineForFramework(runningFramework);
+            settings.LaunchOptions = launchOptions | DebugLaunchOptions.StopDebuggingOnEnd;
+            if(settings.Environment.Count > 0)
+            {
+                settings.LaunchOptions = settings.LaunchOptions | DebugLaunchOptions.MergeEnvironment;
+            }
+            return settings;
+        }
+
+        private async Task<Tuple<string, string>> GetRunnableProjectInformationAsync()
+        {
+            var properties = ConfiguredProject.Services.ProjectPropertiesProvider.GetCommonProperties();
+
+            // Assumes dotnet.exe is on the the path. This will be replaced by accessing a property from the project file
+            string executable, arguments;
+            var targetPath = await properties.GetEvaluatedPropertyValueAsync("TargetPath").ConfigureAwait(false);
+            if (targetPath.EndsWith(".exe"))
+            {
+                executable = targetPath;
+                arguments = "";
+            }
+            else
+            {
+                executable = GetFullPathOfExeFromEnvironmentPath("dotnet.exe");  // await properties.GetEvaluatedPropertyValueAsync("ExeToLaunch").ConfigureAwait(false);
+                if (executable == null)
+                {
+                    executable = "dotnet.exe";
+                }
+                arguments = targetPath.QuoteString();
+            }
+
+            return new Tuple<string, string>(executable, arguments);
+        }
+
+        private async Task<Guid> GetDebuggingEngineAsync()
```

Comment:
Rename to `GetDebuggingEngineGuidAsync`. Since the method returns the Guid of the engine having the word would make it easy to understand the code.